middlewares/validators/checkEmail.js

```diff
@@ -0,0 +1,24 @@
+const User = require('../../lib/modelManagers/usermodel');
+
+/**
+ * @description Check if email is already in the database
+ * @param {*} req
+ * @param {*} res
+ * @param {*} next middleware in the stack
+ * @returns {object} res object
+ */
+async function checkEmail(req, res, next) {
+  try {
+    const record = await User.findEmail(req.body.email);
+    if (record.dataValues.email) {
```

Comment:
Doing this is dangerous. In a scenario where no record is found, null will be returned. performing a property lookup on `null` will throw an error. A better approach should be 

`if (record){} `

Reply:
noted, thanks a lot...I never thought of it that way